### DISCUSSION TITLE
NS Panel - Cannot upload tft file - worried the panel could be bricked

### DISCUSSION THREAD

**Discussion by llcooljerry** [Blackymas/NSPanel_HA_Blueprint#846] [General]
Hello All,

Apologies for Long Post.

**TL;DR - Tried to setup NS Panel, Flashed trough ESPHome and not able to upload TFT file. Tried through Tasmota as well, but was unsuccessful**

I am pretty much a rookie and had been learning quite a lot on my HA journey. But It seems the NS Panel had got me stumped. I went through quite a lot of pages and suggestions to try and resolve the issue.

I am not sure if I messed up and but i need all the advise i can get to get my NS Panel working again.

- NS Panel standard (86/EU Type/120 Type) : White

- Versions
Home Assistant 2023.5.4 (Supervisor 2023.04.1, Frontend 20230503.3 - latest)
ESPHome : 2023.5.4
NSPanel_HA_Blueprint : 3.3

- Installations/Flashing
I am able to successfully flash ESP32 through ESPHome Web, and at the beginning i was unable to flash the nspanel_eu.tft, But found the method to upload a blank tft, which i was able to flash after a few tries. However, I was unable to upload the main eu file even after multiple tries.  'Update TFT display' and even used the 'Exit Reparse' option was used as well. The panel stayed on the blank white screen.

One step i messed up could be, i used Nextion editor to create a blank tft file instead of the file from Github and that was used to upload to the panel.

Since I found few suggestions to try though Tasmota, i flashed Tasmota on the panel through the Web Installer and was able to configure everything except for the steps of flashing the Nextion with the tft files. Tried both blank (from Github) and the main eu file as well but was unsuccessful .

My current screen on the panel is as pictured below.
![20230528NSPANEL](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5267676/f9a25d02-9b41-478a-8f4e-2be88f63ebfe)

In Home Assistant, "Nextion display" is showing 'Unknown' state. However, i was able to find the below details in the ESPHome Wireless log from the panel.
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5267676/b23c5665-fb0b-4aab-8a4b-2c5feeaf5d29)

Currently I am returned with the below errors in the ESPHome Wireless logs while trying to upload the blank or final eu file.
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5267676/9ff9d929-4084-4eae-b119-e1db3856e6a1)
 And
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5267676/1173e964-2b70-43b2-855a-e0ed74a5cf39)

Do let me know if any specific information is needed. I apologize again for the lengthy and cumbersome details for my issue. 

Even in the midst of these, I really do want to Thank @Blackymas , @edwardtfn and all those who supports the project, so that NS Panel have better functionality and features for all DIY users

**Comment by edwardtfn** [comment]
Have you tried using the older Nextion library as described here? https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Common-Issues-TFT-Upload#8-use-an-older-nextion-library-for-esphome

**Reply by llcooljerry** [reply]
Thank you @edwardtfn . I have used the same code after seeing the suggestions and possible solutions in few other threads. 

I updated the device yaml in ESPHome and then tried to install through the 'wireless' method. But it returns the same error messages mentioned above.

Am i required to update older library code in the yaml in ESPHome and then flash it through ESPHome Web (through usb controller) ?

**Reply by edwardtfn** [reply]
> Am i required to update older library code in the yaml in ESPHome and then flash it through ESPHome Web (through usb controller) ?

No. OTA update (via wifi) should be fine.

**Reply by edwardtfn** [reply]
You can try reducing baud_rate.
Just add this to your substitutions:
```yaml 
  baud_rate: "9600"
```

But if this is not fixing the problem, please remove that.

**Reply by edwardtfn** [reply]
If nothing works, you can try transferring the file via serial like in this post: https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/166#issue-1524561202

**Reply by llcooljerry** [reply]
Thank you @edwardtfn . I did try with the 9600 rate but was unsuccessful. I am guessing i would have to follow Kleinheribert's post to check, I'm just not at all experienced with soldering.  But i will try it with maybe the glue method if possible.

**Reply by llcooljerry** [reply]
@edwardtfn is there any chance I, semi bricked ( the relays and switches and temperature all gets reported in the device page as well as the log. Only the screen part is being blank.) the panel ?? 
Was wondering if there was any way to restore the whole panel to Factory settings.

**Comment by edwardtfn** [comment]
And by the way, if you paste that url (http://192.168.1.70:8123/local/nspanel_blank.tft) to your browser (in your computer), do you have the file properly downloaded and with the right file size when compared to the one from GitHub?

**Reply by llcooljerry** [reply]
Yup, after realizing there was file in GitHub i used that for all other flashing. But my initial flash (which got the panel to the white screen) was with my own file from the Nextion editor. 

The file downloads without any errors and the file size is 258 KB and i believe it does match the GitHub file as well.

**Reply by rleongcs** [reply]
I am not sure if you have tried. But killing power all together to the panel and let it restart has heloed me few times in situation where the tft file upload goes bad. Two days back I tried the latest Dev branch and tft upload got stuck at 30% initially and then I was not able to upload after that having the same error as you. Restarting puts the screen into some error which will allow uploading of blank tft and then ultimately the actual tft.

**Reply by llcooljerry** [reply]
Thank you @rleongcs . I did try to turn off and turn on the panel few times and pressing install as soon it shows online on ESPHome.  Most times it shows the same error and one instance it flickers or reboots on its own. I had tried both the blank and eu tft file through this method. But will try it few more times before i move on to the solution mentioned by @edwardtfn.

**Comment by plasterer41** [comment]
You shouldn’t need to do that with soldering etc. I encountered a similar problem and it eventually worked using blackymas blueprint and tft. Delete everything and try from the start again is my advice. Worked for me. Sent from my iPhoneOn 28 May 2023, at 12:22, llcooljerry ***@***.***> wrote:﻿
Thank you @edwardtfn . I did try with the 9600 rate but was unsuccessful. I am guessing i would have to follow Kleinheribert's post to check, I'm just not at all experienced with soldering.  But i will try it with maybe the glue method if possible.

**Reply by llcooljerry** [reply]
Thank you @plasterer41. I did do a fresh flash and install of everything today morning. The ns panel has been keeping me up for 2 days 😅 . But will do another fresh try. There could  be a small step I might be missing.

**Comment by plasterer41** [comment]
Yes, delete everything from home assistant as well and start again. You will get there. 👍🏼Sent from my iPhoneOn 28 May 2023, at 13:47, llcooljerry ***@***.***> wrote:﻿
Thank you @plasterer41. I did do a fresh flash and install of everything today morning. The ns panel has been keeping me up for 2 days 😅 . But will do another fresh try. There could  be a small step I might be missing.

**Reply by llcooljerry** [reply]
Thanks @plasterer41 . I will try out the fresh install as well as the power ON/OFF method and report back.

**Comment by llcooljerry** [comment]
Hello All,

It is with great delight and relief, the panel has now been flashed and the EU tft finally uploaded. I am not exactly sure on what helped because I was trying all your suggestions.

**TL;DR - If you are having issues with uploading tft file, try the following**
- [ ] **Remove everything and Fresh install everything back**
- [ ] **Try cutting power to the panel and let it restart a few times before pressing the Update TFT display button**
- [ ] **make sure the device yaml has the baud rate 9600 for uploading blank tft file** 
- [ ] **Once the blank tft (current one should display QR code) is uploaded, change the baud rate to 115200 for uploading final tft file**

To recap the successful flash/upload;

- I had removed everything and installed everything back - Fresh install

1. removed ESP device and ESPHome addons in HA, removed ESP device from devices list, removed automation and blueprint then restarted HA instance 
2. added ESPHome addon in HA, added new device in ESP home
3. As mentioned by @edwardtfn, used the older nextion library in the device yaml (https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Common-Issues-TFT-Upload#8-use-an-older-nextion-library-for-esphome) with baud_rate: "9600" under Substitutions
4. Flashed the panel with ESPHome Web with the downloaded 'device-factory.bin' file (through USB controller)
5. After completion, Plugged in panel to the mains and had it pop up in HA devices.
6. Added the blueprint and automation as well.

- Flashing blank.tft

1. made sure to have the current nspanel_blank.tft and nspanel_eu.tft in the 'config/www/' folder (/local/nspanel_blank.tft in the yaml). also made sure to use the Home Assistant address and avoiding HTTPS
2. By pressing on the 'Update TFT display' button in the device page, Initially I was returned with the same errors mentioned in the my initial post, but as @rleongcs mentioned to cut off power and then restart the panel, i did the same by turning off and on the mains (since i had it plugged to a socket) 2-3 times with 2 seconds in between.
3. This helped me to complete the flash of the blank.tft (the one with the QR code) to the panel. I was happy and relieved at least the screen was showing the QR code.

- Flashing nspanel_eu.tft

1. Since the blank tft got flashed i was really hopeful and removed baud rate and older nextion library code from the device yaml and corrected the final tft file location as well
2. However, after updating and pressing 'update TFT display' button, it returned with the same errors and the log keep mentioning that the Nextion Display was unavailable and 'Unknown' in the devices page. The log showed that the nextion model and serial numbers etc. to be blank.
3. I did try the cutting power and restart method, but it didn't help in flashing the final eu file.
4. Hence, i reverted the device yaml again to previous one with the baud rate and older nextion library and tried again. But the nextion display was still unavailable / unknown.
5. Per the discussions in the thread (https://github.com/esphome/issues/issues/3519), the user cybericebyte, had mentioned about changing the baud rate to 115200.
6. Once I updated the yaml and updated the device, I ran the update TFT display again. Finally the tft was uploading, though it took some time. But it finally got uploaded to the Panel.

Really want to thank all of you, especially @edwardtfn @rleongcs @plasterer41 with your solutions. I am really grateful for your help. :)

Sorry for the long post, but wanted to share my experience and hoping it helps rookies like me. Thank you everyone.

**Reply by rleongcs** [reply]
Really happy that you finally made it work! 🎉

**Reply by llcooljerry** [reply]
Thank you, @rleongcs. 😊 Thanks to your initial step and using others suggestions, I was able to make it work. Now on to customising the panel. 😁

**Comment by llcooljerry** [comment]
Hello All,
I know Edward is helping cremer with sorting out the upload issue. But just wanted to check with the others @rleongcs , @plasterer41  on whether there is any way **to get camera feeds into the panel** ? 

I been searching quite a bit but didn't find anything on the same, other than adding rtsp links to the orginal sonoff firmware of the panel.

**Reply by rleongcs** [reply]
@llcooljerry Unfortunately there is no way 😞 I did ask when @edwardtfn was surveying in his "[What next?](https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/718#discussioncomment-5723476)" discussion.

**Reply by llcooljerry** [reply]
Damn. That's a bummer. I just found that thread. So only the NS Panel Pro has the ability to stream camera feeds.

Thanks @rleongcs. The panel is still great though, i am still tinkering with the blueprint to add all the entities and buttons. Waiting for my salary to hit so I can support the project. 😅

**Comment by MarkuZP** [comment]
I had the same Issues with updating all of my three NS Panels. Very annoying but it's not a big deal at the end.
The best way to fix this "model does not match" or sometimes "system  error" is described here: https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/197

I make a power cycle of my panels and use the ESP service to flash the blank_tft under developer tools in HA (screenshot in german)
<img width="1016" alt="Bildschirmfoto 2023-09-27 um 11 09 04" src="https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/109298227/638a7ac3-77b9-4aa5-b7a1-19f232f54a13">


1st   -flash the blank tft  (a few seconds)
2nd -restart panel from ESP Home 
<img width="305" alt="Bildschirmfoto 2023-09-27 um 11 15 01" src="https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/109298227/4242ac5f-fb91-47aa-b837-98b386e2a5c0">

3rd - flash the EU_tft (about 10minutes)
Done!

**Reply by brian-kehoe** [reply]
Thank you very much, I was stuck reflashing in ESPHome and trying to use the 'Update TFT button' but to no avail. Your method of calling the service worked first time (although I did have to cut the power to the device after the QR code appeared) and could then use the service to load the nspanel_eu.tft file
